test/unpack_visitor.cpp

```diff
@@ -0,0 +1,135 @@
+#include <msgpack.hpp>
+#include <gtest/gtest.h>
+#include <sstream>
+
+// To avoid link error
+TEST(unpack_visitor, dummy)
+{
+}
+
+#if MSGPACK_DEFAULT_API_VERSION >= 2
+
+struct json_visitor : msgpack::v2::null_visitor {
```

Comment:
Thank you for the comments.

> end_map_value() doesn't append a comma to separate map items like end_array_item() does.

I will fix it.

> visit_str() doesn't process strings as Unicode, therefore non-ASCII codepoints will result in invalid JSON (which encodes them as e.g. "\u00f8").

True. The point of the test and the document is not accurate JSON, I just demonstrate a visitor usage. However, I should use a precise term. I will fix the test and the document. I will use the term 'json like structure ( string is not espaced)'.